test/jdk/java/net/NetworkInterface/UniqueMacAddressesTest.java

```diff
@@ -112,21 +90,20 @@ private String createMacAddressString (NetworkInterface netIf) throws Exception
         return sb.toString();
     }
 
-    private void createNetworkInterfaceList(Enumeration<NetworkInterface> nis,
-            List<NetworkInterface> networkInterfaceList) throws Exception {
-        byte[] macAddr = null;
-        NetworkInterface netIf = null;
-        while (nis.hasMoreElements()) {
-            netIf = (NetworkInterface) nis.nextElement();
-            if (netIf.isUp()) {
-                macAddr = netIf.getHardwareAddress();
-                if (macAddr != null) {
-                    System.out.println("Adding NetworkInterface "
-                            + netIf.getName() + " with mac address "
-                            + createMacAddressString(netIf));
-                    networkInterfaceList.add(netIf);
-                }
-            }
+    private byte[] getNetworkInterfaceHardwareAddress(NetworkInterface inf) {
+        try {
+            return inf.getHardwareAddress();
+        } catch (SocketException se) {
+            throw new UncheckedIOException(se);
         }
     }
+
+    private List<NetIfPair> createNetworkInterfaceList(NetworkConfiguration netConf) {
+        return netConf.interfaces()
+                .map(netIf -> new NetIfPair(netIf.getName(), getNetworkInterfaceHardwareAddress(netIf)))
+                .collect(Collectors.filtering(netIfPair -> netIfPair.address != null,
+                        Collectors.toCollection(ArrayList::new)));
+    }
+
+    record NetIfPair(String interfaceName, byte[] address) {}
```

Comment:
Nit: I'd move that to line 46. It would avoid having to wonder what `NetIfPair` is until you reach the end of the file.